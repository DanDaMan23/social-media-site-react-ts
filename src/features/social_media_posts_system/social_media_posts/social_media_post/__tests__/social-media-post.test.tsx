import { render } from "@testing-library/react"
import SocialMediaPost from "../social-media-post"

describe("social-media-posts test", () => {
  it("should render correctly", () => {
    const { container } = render(
      <SocialMediaPost
        user='mockUser'
        body='mockBody'
        date_created={new Date("2023-01-01")}
      />
    )

    expect(container).toMatchSnapshot()
  })

  it("should render correctly with custom styles", () => {
    const { container } = render(
      <SocialMediaPost
        user='mockUser'
        body='mockBody'
        date_created={new Date("2023-01-01")}
        style={{ color: "red" }}
      />
    )

    expect(container).toMatchSnapshot()
  })
})
