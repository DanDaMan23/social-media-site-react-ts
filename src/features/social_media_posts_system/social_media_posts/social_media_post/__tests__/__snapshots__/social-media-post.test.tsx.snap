// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`social-media-posts test should render correctly 1`] = `
<div>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        mockUser
      </div>
      <p
        class="card-text"
      >
        mockBody
      </p>
      <small>
        December 31, 2022 6:00 PM
      </small>
    </div>
  </div>
</div>
`;

exports[`social-media-posts test should render correctly with custom styles 1`] = `
<div>
  <div
    class="card"
    style="color: red;"
  >
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      >
        mockUser
      </div>
      <p
        class="card-text"
      >
        mockBody
      </p>
      <small>
        December 31, 2022 6:00 PM
      </small>
    </div>
  </div>
</div>
`;
